pyart/aux_io/rainbow_wrl.py

```diff
@@ -0,0 +1,493 @@
+    moving_angle = np.empty(total_rays, dtype='float64')
+    static_angle = np.empty(total_rays, dtype='float64')
+    time_data = np.empty(total_rays, dtype='float64')
+    fdata = np.ma.zeros((total_rays, nbins), dtype='float32',
+                        fill_value=get_fillvalue())
+
+    # read data from file
+    if bfile.endswith('.vol') or bfile.endswith('.azi'):
+        scan_type = 'ppi'
+        sweep_mode['data'] = np.array(nslices * ['azimuth_surveillance'])
+    else:
+        scan_type = 'rhi'
+        sweep_mode['data'] = np.array(['elevation_surveillance'])
+
+    # read data from file:
+    for i in range(nslices):
+        if single_slice:
+            slice_info = common_slice_info
+        else:
+            slice_info = rbf['volume']['scan']['slice'][i]
+
+        # fixed angle
+        t_fixed_angle[i] = float(slice_info['posangle'])
+
+        # fixed angle (repeated for each ray)
+        static_angle[ssri[i]: seri[i]+1] = t_fixed_angle[i]
+
+        # moving angle
+        moving_angle[ssri[i]: seri[i]+1], angle_start, angle_stop = (
+            get_angle(slice_info['slicedata']['rayinfo'],
+                      angle_step=angle_step, scan_type=scan_type))
+
+        # time
+        time_data[ssri[i]:seri[i]+1], sweep_start_epoch = (
+                get_time(slice_info['slicedata']['@date'],
+                         slice_info['slicedata']['@time'],
+                         angle_start[0], angle_stop[-1], angle_step,
+                         rays_per_sweep[i], ant_speed, scan_type=scan_type))
+
+        if i == 0:
+            volume_start_epoch = sweep_start_epoch + 0.
+            start_time = (
+                datetime.datetime.utcfromtimestamp(volume_start_epoch))
+
+        # data
+        fdata[ssri[i]:seri[i]+1, :] = get_data(
+            slice_info['slicedata']['rawdata'],
+            rays_per_sweep[i], nbins)
+
+    if bfile.endswith('.vol') or bfile.endswith('.azi'):
+        azimuth['data'] = moving_angle
+        elevation['data'] = static_angle
+    else:
+        azimuth['data'] = static_angle
+        elevation['data'] = moving_angle
+
+    fixed_angle['data'] = t_fixed_angle
+
+    _time['data'] = time_data-volume_start_epoch
+    _time['units'] = make_time_unit_str(start_time)
+
+    # fields
+    fields = {}
+    # create field dictionary
+    field_dic['_FillValue'] = get_fillvalue()
+    field_dic['data'] = fdata
+    fields[field_name] = field_dic
+
+    # metadata
+    # metadata['instrument_name'] = radar_id
+
+    # instrument_parameters
+    instrument_parameters = dict()
+    instrument_parameters.update({'frequency': frequency})
+
+    return Radar(_time, _range, fields, metadata, scan_type, latitude,
+                 longitude, altitude, sweep_number, sweep_mode, fixed_angle,
+                 sweep_start_ray_index, sweep_end_ray_index, azimuth,
+                 elevation, instrument_parameters=instrument_parameters)
+
+
+def get_angle(ray_info, angle_step=None, scan_type='ppi'):
```

Comment:
Would it make sense to have this function be private, typically this is indicated by prefixing the function with an underscore, `_get_angle`.